app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,68 @@
+class RentalsController < ApplicationController
+  def checkout
```

Comment:
I suggest moving some of the business logic into the models, these methods are quite long.